src/libraries/System.Private.CoreLib/src/System/Action.cs

```diff
@@ -6,37 +6,196 @@
+        where T14 : allows ref struct;
+
+    public delegate void Action<in T1, in T2, in T3, in T4, in T5, in T6, in T7, in T8, in T9, in T10, in T11, in T12, in T13, in T14, in T15>(T1 arg1, T2 arg2, T3 arg3, T4 arg4, T5 arg5, T6 arg6, T7 arg7, T8 arg8, T9 arg9, T10 arg10, T11 arg11, T12 arg12, T13 arg13, T14 arg14, T15 arg15)
+        where T1 : allows ref struct
+        where T2 : allows ref struct
+        where T3 : allows ref struct
+        where T4 : allows ref struct
+        where T5 : allows ref struct
+        where T6 : allows ref struct
+        where T7 : allows ref struct
+        where T8 : allows ref struct
+        where T9 : allows ref struct
+        where T10 : allows ref struct
+        where T11 : allows ref struct
+        where T12 : allows ref struct
+        where T13 : allows ref struct
+        where T14 : allows ref struct
+        where T15 : allows ref struct;
+
+    public delegate void Action<in T1, in T2, in T3, in T4, in T5, in T6, in T7, in T8, in T9, in T10, in T11, in T12, in T13, in T14, in T15, in T16>(T1 arg1, T2 arg2, T3 arg3, T4 arg4, T5 arg5, T6 arg6, T7 arg7, T8 arg8, T9 arg9, T10 arg10, T11 arg11, T12 arg12, T13 arg13, T14 arg14, T15 arg15, T16 arg16)
+        where T1 : allows ref struct
+        where T2 : allows ref struct
+        where T3 : allows ref struct
+        where T4 : allows ref struct
+        where T5 : allows ref struct
+        where T6 : allows ref struct
+        where T7 : allows ref struct
+        where T8 : allows ref struct
+        where T9 : allows ref struct
+        where T10 : allows ref struct
+        where T11 : allows ref struct
+        where T12 : allows ref struct
+        where T13 : allows ref struct
+        where T14 : allows ref struct
+        where T15 : allows ref struct
+        where T16 : allows ref struct;
 
     // This should probably technically be T? rather than T to match `IComparer<T>`.  However, the use cases are generally different,
     // with Comparison<T> typically being used via a lambda, with the T inferred from the type of the collection being sorted,
     // and forcing nullable warnings onto all such usage leads to many spurious warnings.
-    public delegate int Comparison<in T>(T x, T y);
+    public delegate int Comparison<in T>(T x, T y)
+        where T : allows ref struct;
 
-    public delegate TOutput Converter<in TInput, out TOutput>(TInput input);
+    public delegate TOutput Converter<in TInput, out TOutput>(TInput input)
+        where TInput : allows ref struct
+        where TOutput : allows ref struct;
 
-    public delegate bool Predicate<in T>(T obj);
+    public delegate bool Predicate<in T>(T obj)
+        where T : allows ref struct;
 }
 
 namespace System.Buffers
 {
-    public delegate void SpanAction<T, in TArg>(Span<T> span, TArg arg);
-    public delegate void ReadOnlySpanAction<T, in TArg>(ReadOnlySpan<T> span, TArg arg);
+    public delegate void SpanAction<T, in TArg>(Span<T> span, TArg arg)
```

Comment:
It might be nice if we could point users away from `SpanAction` in the future and instead direct them to use `Action<Span<T>, TArg>` instead. It's mostly a point in time thing from this point forward.

Reply:
Similar to how we have "treat [Obsolete] as Error", we need a "Treat Obsolete as Info" for this use-case.

It's not obsolete per se, but it's been made redundant by a summer API

Reply:
If we have delegate type equivalence, we can just make `SpanAction` effectively an alias for `Action<T, Span<TArg>>`. This can also solve other divisions like `Predicate<T>`.

Reply:
I'm not concerned about it. Just looking around GitHub / grep.app, there's basically no usage of it today outside of use of string.Create. I would not expect that to increase, and even if it did, it's not a big deal.

Reply:
I think it would lessen the mental overload if it was - at the very least - removed from Intellisense; EB never should be considered I think.

Reply:
It's used in string.Create. Hiding it from IntelliSense would make it harder to use that.